src/rgw/Makefile

```diff
@@ -0,0 +1,32 @@
+MOTR_SRC_DIR="../../../cortx-motr"
+
+.PHONY: all plugin clean help
+
+all help:
+	@printf "Usage: make COMMAND\n"
+	@printf "\n"
+	@printf "Makefile builds RGW ADDB plugin for m0addb2dump utility."
+	@printf " It allows to detect RGW ADDB probes and parse them to textual "
+	@printf "ADDB dumps for further processing.\n"
+	@printf "\n"
+	@printf "Commands:\n"
+	@printf "Build RGW ADDB plugin:\n"
+	@printf "\tmake plugin\n"
+	@printf "Remove RGW ADDB plugin:\n"
+	@printf "\tmake clean\n"
+	@printf "Print this message:\n"
+	@printf "\tmake\n"
+	@printf "\tmake help\n"
+	@printf "\n"
+
+plugin: rgw_addb_plugin.so
```

Comment:
ADDB is Motr-backend-specific, right? So, I think, all ADDB-related code should just go to rgw_sal_motr.cc.

Reply:
Right, ADDB is Motr-specific, and rgw_sal_motr.cc is the single file in RGW source code tree where ADDB probes were added. There are couple of header files added (rgw_addb.h, rgw_addb_map.h) to keep the code modular. 

Except one thing - unlike Motr, other components that use ADDB probes has to provide so-called ADDB plugin for proper ADDB records parsing. ADDB plugin cannot be placed in rgw_sal_motr.cc file, as it shall be a stand-alone .so binary. Makefile and couple of source files were implemented, but NOT included into the RGW build tree, and proper handling will be tracked in other ticket I about to create for RGW team.

Reply:
> ADDB plugin cannot be placed in rgw_sal_motr.cc file, as it shall be a stand-alone .so binary.

Why?

Reply:
Hmm.... I'm afraid my counter-question to your recent question will be "How?" :)
Okay, we might have been having different points in mind. My point is that rgw_addb_plugin.so is supplied to m0addb2dump utility as an input (contains necessary parsers) and m0addb2dump takes probes translation table and makes text dump. 
If we relied on standard RGW build (saying, no Makefile, no addb .so), rgw_sal_motr.cc will be compiled to .o file, but still need to make a shared library from it - and extra step is needed, as m0addb2dump still requires .so file.
Have I understand you correctly that you suggest to make a shared-lib from rgw_sal_motr.cc or what?

Reply:
No, I just wasn't aware that m0addb2dump requires .so. I'm still not sure I understand - why is that so. Why can't it cake radosgw binary as an input?

Reply:
Honestly speaking, I haven't tried it and have used a well-known approach developed for S3 server application. 
Is there a strict requirement that no additional files (like plugin .so) shall be added to RGW codebase?

Reply:
Conducted an experiment to substitute plugin.so with radosgw binary, and it doesn't work out-of-box. I expect radosgw binary has to be built with specific compiler and linker options (-fPIC, export symbol table, etc.) to behave simultaneously as a binary executable and a shared library. 
m0addb2dump in its turn expects shared library and opens it with dlopen() system call.

Reply:
> Is there a strict requirement that no additional files (like plugin .so) shall be added to RGW codebase?

@maximalezhin we can add additional files, but we need create a store specific dir and put all extra files there. For motr, please create dir `/src/rgw/store/motr` and move the newly added files there. 
Please check [dbstore](https://github.com/ceph/ceph/tree/master/src/rgw/store/dbstore) for reference.

Reply:
> ...used a well-known approach developed for S3 server application.

Is it required from any motr application to have .so in order to use addb? cc @nikitadanilov

Reply:
m0addb2dump is a utility that parses binary addb stob and produces human-readable addb output. m0addb2dump has a plugin system that allows non-motr addb producers to register parsers for new addb records. So if rgw produces new types of addb records (which it probably does) and we want that standard m0addb2dump would be able to parse there records, rgw should provide a plugin with new parsers.

Reply:
Thank you, Nikita, commenting on this. cc: @nikitadanilov 
@andriytk as Nikita said, RGW produces new types of addb records. We don't want to include those types in motr sources (because they are not motr-related), but still want to support them. In this case, plugin, in the form of shared library, is provided.

Reply:
I see...

@maximalezhin, please move this plugin thing to `src/rgw/motr/addb/` directory.

Reply:
@andriytk plugin moved to motr/addb. Would you plz review?